src/typing_extensions/ext.py

```diff
@@ -0,0 +1,99 @@
+"""
+Protocols and type aliases that are not scheduled for inclusion in typing.
+"""
+
+from os import PathLike
+from typing import AbstractSet, Awaitable, Container, Iterable, Tuple, TypeVar, Union
+from typing_extensions import Protocol, TypeAlias
+
+_T_co = TypeVar("_T_co", covariant=True)
+_T_contra = TypeVar("_T_contra", contravariant=True)
+_KT = TypeVar("_KT")
+_KT_co = TypeVar("_KT_co", covariant=True)
+_KT_contra = TypeVar("_KT_contra", contravariant=True)
+_VT_co = TypeVar("_VT_co", covariant=True)
+
+#
+# Protocols for dunder methods
+#
+
+
+class SupportsAnext(Protocol[_T_co]):
+    def __anext__(self) -> Awaitable[_T_co]:
+        ...
+
+
+class SupportsDivMod(Protocol[_T_contra, _T_co]):
+    def __divmod__(self, __other: _T_contra) -> _T_co:
+        ...
+
+
+class SupportsGetItem(Container[_KT_contra], Protocol[_KT_contra, _VT_co]):
+    def __getitem__(self, __k: _KT_contra) -> _VT_co:
+        ...
+
+
+class SupportsNext(Protocol[_T_co]):
+    def __next__(self) -> _T_co:
+        ...
+
+
+class SupportsRDivMod(Protocol[_T_contra, _T_co]):
```

Comment:
This seems very obscure, and Python only uses it in two places. I think we can skip it.

In fact even `SupportsDivMod` (without the R) seems marginal; it's also only used twice in typeshed.

Reply:
But it *is* used. I think it doesn't hurt to have them here instead of `typing`. In fact, for completeness and consistency's sake I think it will make sense in the long run to have protocols for the other arithmetic operators here.

Reply:
I think `SupportsRDivMod` is _okay_, though I agree with @JelleZijlstra that it isn't really used much, so I'd also be fine with omitting it.

A lot of the other arithmetic dunders are quite tricky, though (BinOp dunders in general are quite tricky), due to the presence of the reflected dunders. It can be hard to predict when you _actually_ need an object with an `__add__` method, and when "anything with either an `__add__` or an `__radd__` method" will do. `SupportsRichComparison` in `_typeshed` is a really good example: it has to be a union of protocols to work and, although it works really well for loads of functions in typeshed, we still have to use a _different_ union of protocols for the functions in `_operator.pyi`.